draco/synthesis/noise.py

```diff
@@ -62,6 +68,59 @@ def process(self, data):
         return data
 
 
+class GaussianNoiseDataset(task.SingleTask):
+    """Generates a Gaussian distributed noise dataset using the
+    the noise estimates of an existing dataset.
+
+    Attributes
+    ----------
+    seed : int
+        Random seed for the noise generation.
+    """
+
+    seed = config.Property(proptype=int, default=None)
+
+    def process(self, data):
+        """Generates a Gaussian distributed noise dataset,
+        given the provided dataset's visibility weights
+
+        Parameters
+        ----------
+        data : :class:`VisContainer`
+            Any dataset which contains a vis and weight attribute.
+            Note the modification is done in place.
+
+        Returns
+        -------
+        data_noise : same as :param:`data`
+            The previous dataset with the visibility replaced with
+            a Gaussian distributed noise realisation.
+
+        """
+        # Distribute in something other than `stack`
+        data.redistribute("freq")
+        # Visibility to be replaced by noise
+        vis = data.vis[:]
+
+        # create a random generator, and create a local seed state
+        rg = randomgen.generator.RandomGenerator()
+
+        with randomgen_mpi_random_seed(rg, self.seed) as rg:
+            noise = rg.normal(
```

Comment:
This looks to me like it will fail `black`. Has it been run through it?

Reply:
It has now!